mysql-test/suite/encryption/t/percona_parallel_dblwr_encrypt.test

```diff
@@ -0,0 +1,331 @@
+CREATE TABLE t2(a TEXT);
+CREATE TABLE t3(a TEXT) ENCRYPTION='N';
+
+SET GLOBAL innodb_limit_optimistic_insert_debug=2;
+SET GLOBAL DEFAULT_TABLE_ENCRYPTION=ONLINE_TO_KEYRING;
+SET GLOBAL innodb_encryption_threads=4;
+
+--let tables_count=`select count(*) from INFORMATION_SCHEMA.INNODB_TABLESPACES`
+--echo # Wait max 10 min for key encryption threads to encrypt all spaces
+--let $wait_timeout= 600
+# - 2 we do not encrypt t3 and temporary tablespace
+--let $wait_condition=SELECT COUNT(*) = $tables_count - 2 FROM INFORMATION_SCHEMA.INNODB_TABLESPACES_ENCRYPTION WHERE MIN_KEY_VERSION = 1
+--source include/wait_condition.inc
+
+SET GLOBAL innodb_encryption_threads=0;
+SET GLOBAL DEFAULT_TABLE_ENCRYPTION=OFF;
+
+--echo # From now on t1 and t2 should be encrypted
+--echo # t3 should remain unencrypted.
+
+SET GLOBAL innodb_parallel_dblwr_encrypt=ON;
+SET GLOBAL innodb_buf_flush_list_now=ON;
+SET GLOBAL innodb_checkpoint_disabled=ON;
+
+INSERT INTO t1 (a) VALUES ('Abracadabra in t1 should be encrypted in dblwr buffer');
+INSERT INTO t2 (a) VALUES ('Abracadabra in t2 should be encrypted in dblwr buffer');
+INSERT INTO t3 (a) VALUES ('Abracadabra in t3 should NOT be encrypted in dblwr buffer');
+
+# Wait for all dirty pages to be flushed.
+--let $wait_condition= SELECT variable_value = 0 FROM performance_schema.global_status WHERE LOWER(variable_name) = 'innodb_buffer_pool_pages_dirty'
+--source include/wait_condition.inc
+
+--source include/kill_mysqld.inc
+
+--echo # Writes to Parallel dblwr are encrypted, so the pattern should be not found
+--let SEARCH_PATTERN=in t1 should be encrypted
+--let SEARCH_FILE=$MYSQLD_DATADIR/#ib_16384_*
+--let ABORT_ON=FOUND
+--source include/search_pattern_in_file.inc
+
+--echo # Writes to Parallel dblwr are encrypted, so the pattern should be not found
+--let SEARCH_PATTERN=in t2 should be encrypted
+--source include/search_pattern_in_file.inc
+
+--echo # Writes to Parallel dblwr are encrypted, but t3 is not encrypted, so the pattern should be found
+--let ABORT_ON=NOT_FOUND
+--let SEARCH_PATTERN=in t3 should NOT be encrypted
+--source include/search_pattern_in_file.inc
+
+--let $restart_parameters = "restart: --debug=d,force_dblwr_decryption"
+--source include/start_mysqld.inc
+
+# t1 is now encrypted, insert more data into it, as we will abort decryption
+# after decrypting first 100 pages in t1
+delimiter //;
+create procedure innodb_insert_proc (repeat_count int)
+begin
+  declare current_num int;
+  set current_num = 0;
+  while current_num < repeat_count do
+    insert into t1 values (repeat('foobar',42));
+    set current_num = current_num + 1;
+  end while;
+end//
+delimiter ;//
+commit;
+
+set autocommit=0;
+call innodb_insert_proc(30000);
+commit;
+set autocommit=1;
+
+SET GLOBAL DEFAULT_TABLE_ENCRYPTION=ONLINE_FROM_KEYRING_TO_UNENCRYPTED;
+SET GLOBAL innodb_encryption_threads=4;
+
+--echo # We want only first 100(x=100 by default) pages to be rotated
+SET GLOBAL debug="+d,rotate_only_first_x_pages_from_t1";
```

Comment:
Did we miss SET GLOBAL innodb_parallel_dblwr_encrypt=ON; If not please add it.. (WIth OFF, I guess it tested by other kill and restart scenarios)